### PR TITLE
feat(dropdown): add disabled to dropdown

### PR DESCRIPTION
**Pull request by paghar**
#539

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change contains documentation update

## Breaking changes

Please document the breaking changes if suitable.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by tulup-conner** [comment]
Please add at least one test case for this verifying the disabled behavior.

**Comment by paghar** [comment]
@tulup-conner  I added test case. Please check it.

**Review by tulup-conner — CHANGES_REQUESTED** [review]
Thank you! Can you rebase and move these changes to src/theme.ts? We just changed up our repository pretty substantially. You can see the new contributing guide here: https://github.com/themesberg/flowbite-react/blob/main/CONTRIBUTING.md

If you aren't comfortable rebasing, let me know and I'll be happy to do so for you.

**Comment by paghar** [comment]
@tulup-conner I will do it.

File: src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -70,6 +70,7 @@ const DropdownComponent: FC<DropdownProps> = ({
     inline,
     floatingArrow = false,
     arrowIcon = true,
+    disabled,
```
[the remark this answers is not in the document]
Comment (paghar) [reply src/components/Dropdown/Dropdown.tsx]: Done

File: src/components/Floating/Floating.tsx [hunk 2]
```diff
@@ -65,6 +66,7 @@ export const Floating: FC<FloatingProps> = ({
   theme,
   trigger = 'hover',
   minWidth,
+  disabled,
```
[the remark this answers is not in the document]
Comment (paghar) [reply src/components/Floating/Floating.tsx]: @rluders Done

**Review by rluders — APPROVED** [review]
lgtm

**Comment by rluders** [comment]
@tulup-conner up to you know.

**Comment by rluders** [comment]
@paghar can you rebase this one with `main` branch? I think that there are some conflicts to be solved. Once done, ping me than I can get it merged. Thanks.

**Comment by rluders** [comment]
@paghar Please, revert your merge commit and use rebase instead of merge to update your branch.

**Comment by tulup-conner** [comment]
Let me know when this has been updated and I will give it another pass

**Comment by rluders** [comment]
This was completed by #840

**Pull request closed without merging**